File: Assessment/Assessment.Entities/SubjectService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class SubjectService
    {
        private ISubjectRepository _repository;

        public SubjectService(ISubjectRepository repository)
        {
            _repository = repository;
        }

        public void Insert(Subject subject)
        {
            _repository.Insert(subject);
        }

        public void Delete(int subjectID)
        {
            _repository.Delete(subjectID);
        }

        public void Update(Subject subject)
        {
            _repository.Update(subject);
        }

        public Subject Get(int subjectID)
        {
            return _repository.Get(subjectID);
        }

        public IEnumerable<Subject> GetAll()
        {
            return _repository.GetAll();
        }
    }
}

File: Assessment/Assessment.WEB/src/app/services/subject.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Subject } from '../entities/subject';

export class SubjectService {
    http: Http;
    baseUrl: string;
    controller: string = "api/Subject/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public update(subject: Subject) {
        return this.http.put(this.baseUrl + this.controller, subject);
    }

    public insert(subject: Subject) {
        return this.http.post(this.baseUrl + this.controller, subject);
    }

    public delete(id: number) {
        return this.http.delete(this.baseUrl + this.controller + id);
    }
}